hi-ml-histopathology/src/histopathology/configs/classification/DeepSMILEPanda.py

```diff
@@ -94,7 +116,171 @@ def get_data_module(self) -> PandaTilesDataModule:
+                                                 dropout_rate=self.dropout_rate,
+                                                 class_weights=self.data_module.class_weights,
+                                                 l_rate=self.l_rate,
+                                                 weight_decay=self.weight_decay,
+                                                 adam_betas=self.adam_betas,
+                                                 is_finetune=self.is_finetune,
+                                                 class_names=self.class_names,
+                                                 outputs_handler=outputs_handler,
+                                                 chunk_size=self.encoding_chunk_size,
+                                                 n_epochs=self.max_epochs)
+        outputs_handler.set_slides_dataset(self.get_slides_dataset())
+        return deepmil_module
+
+    def get_encoder(self) -> TileEncoder:
+        # Myronenko et al. 2021 uses Resnet50 CNN encoder
+        if self.encoder_type == ImageNetEncoder.__name__:
+            return ImageNetEncoder_Resnet50(feature_extraction_model=resnet50,
+                                            tile_size=self.tile_size, n_channels=self.n_channels)
+
+        elif self.encoder_type == ImageNetSimCLREncoder.__name__:
+            return ImageNetSimCLREncoder(tile_size=self.tile_size, n_channels=self.n_channels)
+
+        elif self.encoder_type == HistoSSLEncoder.__name__:
+            return HistoSSLEncoder(tile_size=self.tile_size, n_channels=self.n_channels)
+
+        elif self.encoder_type == SSLEncoder.__name__:
+            return SSLEncoder(pl_checkpoint_path=self.downloader.local_checkpoint_path,
+                              tile_size=self.tile_size, n_channels=self.n_channels)
+
+        else:
+            raise ValueError(f"Unsupported encoder type: {self.encoder_type}")
+
+    def get_pooling_layer(self) -> Tuple[nn.Module, int]:
+        # Myronenko et al. 2021 has a different transformer pooling layer.
+        # This is defined in `TransformerPooling`.
+        num_encoding = self.encoder.num_encoding
+
+        pooling_layer: nn.Module
+        if self.pool_type == AttentionLayer.__name__:
+            pooling_layer = AttentionLayer(num_encoding,
+                                           self.pool_hidden_dim,
+                                           self.pool_out_dim)
+        elif self.pool_type == GatedAttentionLayer.__name__:
+            pooling_layer = GatedAttentionLayer(num_encoding,
+                                                self.pool_hidden_dim,
+                                                self.pool_out_dim)
+        elif self.pool_type == MeanPoolingLayer.__name__:
+            pooling_layer = MeanPoolingLayer()
+        elif self.pool_type == MaxPoolingLayer.__name__:
+            pooling_layer = MaxPoolingLayer()
+        elif self.pool_type == TransformerPooling.__name__:
```

Comment:
`TransformerPooling` is repeated here, but we have two different transformer pooling layers in cpath. We need to rename one of these.

Reply:
I think is MaxPoolingLayer that is repeated